### PR TITLE
add bypass index and type

### PR DESCRIPTION
**Pull request by nnovikov**
I add some changes for bypass 'index' and 'type' fileds directly to elasticsearch. You need logstash_format=false in your config.

### CODE REVIEW

File: lib/fluent/plugin/out_elasticsearch.rb [hunk 1]
```diff
@@ -165,15 +165,19 @@ def write(chunk)
         else
           target_index = "#{@logstash_prefix}-#{Time.at(time).strftime("#{@logstash_dateformat}")}"
         end
+        target_type = @type_name
       else
-        target_index = @index_name
+	target_index = record.fetch("_index", @index_name)
```
Comment (pitr) [inline lib/fluent/plugin/out_elasticsearch.rb]: fix weird spacing

**Comment by pitr** [comment]
I think this behaviour needs to be enabled by a toggle setting, off by default. That way existing users would not be impacted.

**Comment by nnovikov** [comment]
I add this PR as example for this issue https://github.com/uken/fluent-plugin-elasticsearch/issues/149
If you approve this feature, I will add config parameter (what name you seem good?) and fix spacing. Also if you have any additional request, please tell me.

**Comment by pitr** [comment]
Definitely approve. How about `index_name_from_record` with value signifying the key on record (eg. "_index"), default being `nil` in which case disabling this feature. And similarly, `type_name_from_record`.

Would be nice to have a test for each, and an entry in README

**Comment by nnovikov** [comment]
I try to write test, but have error:
/usr/lib/ruby/2.1.0/rubygems/core_ext/kernel_require.rb:55:in `require': cannot load such file -- fluent/test (LoadError)
I need install fluentd? Or I can test module only?

**Comment by pitr** [comment]
You need to run `bundle install` to get all dependencies

**Comment by pitr** [comment]
implemented in #169

**Pull request closed without merging**